src/foam/nanos/u2/navigation/UserInfoNavigationView.js

```diff
@@ -70,17 +70,31 @@ foam.CLASS({
                 .add( this.subject.realUser.toSummary() )
               .end();
         }));
-      var menuArray = (await menu.children.orderBy(foam.nanos.menu.Menu.ORDER).select())
-        .array.map( menuItem => {
-          var e = this;
-          return self.Action.create({
-            name: menuItem.name,
-            label: this.translationService.getTranslation(foam.locale, menuItem.id + '.label', menuItem.label),
-            code: () => {
-              menuItem.launch_(X, e);
-            }
-          });
+
+      // We need to add menus from settings (and then add menus from theme.settingsRootMenu) 
+      // because some menus are used in both settings and theme.settingsRootMenu (e.g., sign-out).
+      // Doing this prevents us from creating the same menu for each setting.
+      let menu = this.Menu.create({ id: 'settings' });
```

Comment:
why are we not finding this menu? why create?

Reply:
code between lines 77- 86 aren't very clear to me. To me the below makes more sense.
`
let menuId = this.theme.settingsRootMenu || 'settings';
let menu = this.menuDAO.find(menuId);
let menuArray = (await menu.children..orderBy(foam.nanos.menu.Menu.ORDER).select()).array`

on line 83 you are your concatenating 'settings' with what ever is placed as theme.settingsRootMenu ... wouldn't that defeat the purpose of defining another menu for a theme. You'd never be able to adjust the base 'settings' menu list.

Reply:
The logic is like add all the menus in `theme.settingsRootMenu` and menus that you have permission in `settings`. This is needed as some menus in `settings` are also used in different settings (e.g., sign-out is used in `settings` and `smeSettings`). As to not being able to adjust the base 'settings` menu list, we still can do it using permission.

Reply:
hmmm - I see the issue. 
How about something in the middle. Instead of concat the two parent menus, use the theme.settingsRootMenu || 'settings' and then add 'sign-out'.
Remove the parent from sign-out therefore it will never be duplicated.